### PR TITLE
Router: Lazy Loading

### PR DESCRIPTION
**Pull request by vsavkin**

### CODE REVIEW

File: modules/@angular/core/src/linker/systemjs_component_resolver.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+/**
+ * Component resolver that can load components lazily
+ */
+export class SystemJsComponentResolver implements ComponentResolver {
+  constructor(private _resovler: ComponentResolver) {}
+
+  resolveComponent(componentType:string|Type):Promise<ComponentFactory<any>> {
+    if (isString(componentType)) {
+      return (<any>global).System.import(componentType).then(module =>
+        this._resovler.resolveComponent(module.default));
```
Comment (mhevery) [inline modules/@angular/core/src/linker/systemjs_component_resolver.ts]: default is fine for now, but in long term we should be able to enumerate all of the components and then take the one which matches `matchType` or take it if it is the only one.

**Comment by mhevery** [comment]
LGTM

File: modules/@angular/core/src/linker/systemjs_component_resolver.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import { ComponentResolver } from './component_resolver';
+import { Type, isString, global } from '../../src/facade/lang';
+import { ComponentFactory } from './component_factory';
+
+/**
+ * Component resolver that can load components lazily
+ */
+export class SystemJsComponentResolver implements ComponentResolver {
+  constructor(private _resovler: ComponentResolver) {}
```
Comment (frederikschubert) [inline modules/@angular/core/src/linker/systemjs_component_resolver.ts]: Is resovler a typo?

**Comment by vsavkin** [comment]
Merged via https://github.com/angular/angular/commit/b30ddfbfc5192f526ceaac525cf6965635831c46

**Pull request merged**